src/transformations.jl

```diff
@@ -223,84 +283,135 @@ julia> StatsBase.transform(dt, X)
+            tmax = X[i]
+        end
+    end
+    tmax = 1 / (tmax - tmin)
+    return UnitRangeTransform(1, dims, unit, [tmin], [tmax])
 end
 
-function transform!(y::AbstractVecOrMat{<:Real}, t::UnitRangeTransform, x::AbstractVecOrMat{<:Real})
-    d = t.dim
-    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
-    n = size(x,2)
-    size(y,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+function transform!(y::AbstractMatrix{<:Real}, t::UnitRangeTransform, x::AbstractMatrix{<:Real})
+    if t.dims == 1
+        l = t.len
+        size(x,2) == size(y,2) == l || throw(DimensionMismatch("Inconsistent dimensions."))
+        n = size(x,1)
+        size(y,1) == n || throw(DimensionMismatch("Inconsistent dimensions."))
 
-    tmin = t.min
-    tscale = t.scale
+        tmin = t.min
+        tscale = t.scale
 
-    if t.unit
-        broadcast!((x,s,m) -> (x-m)*s, y, x, tscale, tmin)
-    else
-        broadcast!(*, y, x, tscale)
+        if t.unit
+            broadcast!((x,s,m)->(x-m)*s, y, x, tscale', tmin')
+        else
+            broadcast!(*, y, x, tscale')
+        end
+        return y
+    elseif t.dims == 2
+        t_ = UnitRangeTransform(t.len, 1, t.unit, t.min, t.scale)
+        return transform!(y', t_, x')'
     end
-    return y
 end
 
-function reconstruct!(x::AbstractVecOrMat{<:Real}, t::UnitRangeTransform, y::AbstractVecOrMat{<:Real})
-    d = t.dim
-    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
-    n = size(y,2)
-    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+function reconstruct!(x::AbstractMatrix{<:Real}, t::UnitRangeTransform, y::AbstractMatrix{<:Real})
+    if t.dims == 1
+        l = t.len
+        size(x,2) == size(y,2) == l || throw(DimensionMismatch("Inconsistent dimensions."))
+        n = size(y,1)
+        size(x,1) == n || throw(DimensionMismatch("Inconsistent dimensions."))
 
-    tmin = t.min
-    tscale = t.scale
+        tmin = t.min
+        tscale = t.scale
 
-    if t.unit
-        broadcast!((y,s,m) -> y/s + m, x, y, tscale, tmin)
-    else
-        broadcast!(/, x, y, tscale)
+        if t.unit
+            broadcast!((y,s,m)->y/s+m, x, y, tscale', tmin')
+        else
+            broadcast!(/, x, y, tscale')
+        end
+        return x
+    elseif t.dims == 2
+        t_ = UnitRangeTransform(t.len, 1, t.unit, t.min, t.scale)
+        return reconstruct!(x', t_, y')'
     end
-    return x
 end
 
 """
-    standardize(DT, X; kwargs...)
+    standardize(DT, X; dims=nothing, kwargs...)
 
-Return a row-standardized matrix `X` using `DT` transformation which is a subtype of `AbstractDataTransform`:
+ Return a standardized copy of vector or matrix `X` along dimensions `dims`
+ using transformation `DT` which is a subtype of `AbstractDataTransform`:
 
 - `ZScoreTransform`
 - `UnitRangeTransform`
 
+Return a standardized matrix while the input of `X` is a `AbstractMatrix`.
```

Comment:
These two lines sound redundant with the description above.

Reply:
OK.